testtools/iothub_test/src/iothub_account.c

```diff
@@ -749,6 +757,7 @@ IOTHUB_ACCOUNT_INFO_HANDLE IoTHubAccount_Init_With_Config(IOTHUB_ACCOUNT_CONFIG*
                     if (retrieveConnStringInfo(iothub_account_info) != 0)
                     {
                         LogError("retrieveConnStringInfo failed.\r\n");
+                        free(iothub_account_info->sasDevices);
```

Comment:
Aren't x509Certificate and x509PrivateKey also supposed to be freed in all the error blocks below?